src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java

```diff
@@ -0,0 +1,453 @@
+package com.softserve.teamproject.service.impl;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doNothing;
+
+import com.softserve.teamproject.TestData;
+import com.softserve.teamproject.dto.GroupsFilter;
+import com.softserve.teamproject.entity.Group;
+import com.softserve.teamproject.entity.Location;
+import com.softserve.teamproject.entity.User;
+import com.softserve.teamproject.entity.assembler.GroupResourceAssembler;
+import com.softserve.teamproject.entity.resource.GroupResource;
+import com.softserve.teamproject.repository.GroupRepository;
+import com.softserve.teamproject.repository.LocationRepository;
+import com.softserve.teamproject.repository.StatusRepository;
+import com.softserve.teamproject.repository.UserRepository;
+import com.softserve.teamproject.service.TestGroup;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import javax.persistence.EntityManager;
+import javax.validation.ValidationException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.test.context.support.WithUserDetails;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@ActiveProfiles("test")
+public class GroupServiceImplTest {
+
+  private final String TEACHER_LOCATION_WITH_GROUPS = "OlegShvets";
+  private final String TEACHER_WITH_GROUPS = "OlegShvets";
+  private final String COORDINATOR = "DmytroPetin";
+  private final String COORDINATOR_OTHER_LOCATION = "LukasLukichich";
+  private final String TEACHER_WITHOUT_GROUPS = "NogroupsUser";
+  private final String TEACHER_LOCATION_WITHOUT_GROUPS = "NoOlegShvets";
+  @Spy
+  private GroupResourceAssembler groupResourceAssembler;
```

Comment:
I can see that you prefer using spies over mocks. Do you really need spies here?